src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -812,15 +812,26 @@ public void RemoveExtension(IGrainExtension handler)
             return false;
         }
 
+        private ExtensionInvoker GetCurrentExtensionInvoker()
+        {
+            return (RuntimeContext.CurrentActivationContext.ContextType == SchedulingContextType.SystemTarget)
+                ? (RuntimeContext.CurrentActivationContext as SchedulingContext)?.SystemTarget.ExtensionInvoker
+                : GetCurrentActivationData().ExtensionInvoker;
+        }
```

Comment:
These methods should access `RuntimeContext.CurrentActivationContext` only once. It's a thread local, not as cheap as a regular field access.